app/views/common/_discourse_splash.html.erb

```diff
@@ -1,196 +1,79 @@
-          --success: #<%= ColorScheme.hex_for_name("success", scheme_id) %>;
-        }
-        <%- else %>
-        /* user picked a theme a light scheme and also enabled a dark scheme */
-
-        /* deal with light scheme first */
-        @media (prefers-color-scheme: light) {
-          :root {
-            --primary: #<%= ColorScheme.hex_for_name("primary", scheme_id) %>;
-            --secondary: #<%= ColorScheme.hex_for_name("secondary", scheme_id) %>;
-            --tertiary: #<%= ColorScheme.hex_for_name("tertiary", scheme_id) %>;
-            --highlight: #<%= ColorScheme.hex_for_name("highlight", scheme_id) %>;
-            --success: #<%= ColorScheme.hex_for_name("success", scheme_id) %>;
-          }
-        }
-
-        /* then deal with dark scheme */
-        @media (prefers-color-scheme: dark) {
-          :root {
-            --primary: #<%= ColorScheme.hex_for_name("primary", dark_scheme_id) %>;
-            --secondary: #<%= ColorScheme.hex_for_name("secondary", dark_scheme_id) %>;
-            --tertiary: #<%= ColorScheme.hex_for_name("tertiary", dark_scheme_id) %>;
-            --highlight: #<%= ColorScheme.hex_for_name("highlight", dark_scheme_id) %>;
-            --success: #<%= ColorScheme.hex_for_name("success", dark_scheme_id) %>;
-          }
-        }
-        <%- end %>
-
-        /* these styles need to live here because the SVG has a different scope */
-        .dots {
-          animation-name: loader;
-          animation-timing-function: ease-in-out;
-          animation-duration: 3s;
-          animation-iteration-count: infinite;
-          animation-play-state: var(--animation-state);
-          stroke: #fff;
-          stroke-width: 0.5px;
-          transform-origin: center;
-          opacity: 0;
-          r: max(1vw, 11px);
-          cy: 50%;
-          filter: saturate(2) opacity(0.85);
-          fill: var(--tertiary);
-        }
-
-        .dots:nth-child(2) {
-          animation-delay: 0.15s;
-        }
-
-        .dots:nth-child(3) {
-          animation-delay: 0.3s;
-        }
-
-        .dots:nth-child(4) {
-          animation-delay: 0.45s;
-        }
-
-        .dots:nth-child(5) {
-          animation-delay: 0.6s;
-        }
-
-        @keyframes loader {
-          0% {
-            opacity: 0;
-            transform: scale(1);
-          }
-          45% {
-            opacity: 1;
-            transform: scale(0.7);
-          }
-          65% {
-            opacity: 1;
-            transform: scale(0.7);
-          }
-          100% {
-            opacity: 0;
-            transform: scale(1);
-          }
-        }
-      </style>
-
-      <g class="container">
-        <circle class="dots" cx="30vw" />
-        <circle class="dots" cx="40vw" />
-        <circle class="dots" cx="50vw" />
-        <circle class="dots" cx="60vw" />
-        <circle class="dots" cx="70vw" />
-      </g>
+      <!--
+        LCP candidate image
+        #{"." * 5000}
```

Comment:
Would there be any benefit to storing this whole string in a constant somewhere so we can re-use a frozen string here?

Reply:
Yup, good idea. I am also wondering if we can generate this client-side to avoid shipping 5kb of rubbish in our html payloads 😅 (although maybe not too much of a concern, since it'll compress very well)